test/Microsoft.ML.Tests/TrainerEstimators/MatrixFactorizationTests.cs

```diff
@@ -62,7 +63,7 @@ public void MatrixFactorizationSimpleTrainAndPredict()
             var data = reader.Read(new MultiFileSource(GetDataPath(TestDatasets.trivialMatrixFactorization.trainFilename)));
 
             // Create a pipeline with a single operator.
-            var pipeline = new MatrixFactorizationTrainer(mlContext, userColumnName, itemColumnName, labelColumnName,
+            var pipeline = mlContext.Recommendation().Trainers.MatrixFactorization(userColumnName, itemColumnName, labelColumnName,
```

Comment:
>mlContext.Recommendation().Trainers.MatrixFactorization( [](start = 27, length = 56)

why I can do 
`mlContext.BinaryClassification.Trainers.FastTree`
but for recommendaion i need to do 
`mlContext.Recommendation().Trainers`
I'm talking about braces.

Reply:
Because RecommendationContext lives in a different assembly and nuget. So it is an extension method to MLContext, rather than a  property. 

there are no recommendation trainers in ml.net core nuget, no point on having it there and empty. Time series will be the same. 

---
In reply to: [235502233](https://github.com/dotnet/machinelearning/pull/1682#discussion_r235502233) [](ancestors = 235502233)